std/experimental/testing/should.d

```diff
@@ -0,0 +1,561 @@
+module std.experimental.testing.should;
+
+import std.exception;
+import std.conv;
+import std.algorithm;
+import std.traits;
+import std.range;
+
+public import std.experimental.testing.attrs;
+
+@safe:
+
+/**
+An exception to signal that a test case has failed.
+ */
+class UnitTestException: Exception
+{
+    this(in string[] msgLines,
+         in string file = __FILE__,
+         in ulong line = __LINE__,
+         Throwable next = null)
+    {
+        super(msgLines.join("\n"), next, file, line);
+        this.msgLines = msgLines;
+    }
+
+    override string toString() const pure
+    {
+        return msgLines.map!(a => getOutputPrefix(file, line) ~ a).join("\n");
+    }
+
+private:
+
+    const string[] msgLines;
+
+    string getOutputPrefix(in string file, in ulong line) const pure
+    {
+        return "    " ~ file ~ ":" ~ line.to!string ~ " - ";
+    }
+}
+
+/**
+Verify that the condition is `true`.
+Throws: UnitTestException on failure.
+ */
+void shouldBeTrue(E)(lazy E condition,
+                     in string file = __FILE__, in ulong line = __LINE__)
+{
+    if (!condition) failEqual(condition, true, file, line);
+}
+
+unittest
+{
+    assertOk(shouldBeTrue(true));
+}
+
+
+/**
+Verify that the condition is `false`.
+Throws: UnitTestException on failure.
+ */
+void shouldBeFalse(E)(lazy E condition,
+                      in string file = __FILE__, in ulong line = __LINE__)
+{
+    if (condition) failEqual(condition, false, file, line);
+}
+
+unittest
+{
+    assertOk(shouldBeFalse(false));
+}
+
+/**
+Verify that two values are the same.
+Throws: UnitTestException on failure.
+ */
+void shouldEqual(T, U)(in T value, in U expected,
+                       in string file = __FILE__, in ulong line = __LINE__)
+  if (is(typeof(value != expected) == bool) && !is(T == class))
+{
+    if (value != expected) failEqual(value, expected, file, line);
+}
+
+/**
+Verify that two values are the same.
+Throws: UnitTestException on failure
+ */
+void shouldEqual(T)(in T value, in T expected,
+                    in string file = __FILE__, in ulong line = __LINE__)
+if (is(T == class))
```

Comment:
the if constraints are not aligned equal everywhere, try dfmt into a new file and diff